File: sketches/sketches.model/src/main/java/edu/kit/kastel/lissa/sketches/model/elements/uml/IUMLThing.java
package edu.kit.kastel.lissa.sketches.model.elements.uml;

import org.eclipse.collections.api.list.ImmutableList;

import edu.kit.kastel.lissa.sketches.model.elements.IBox;

public interface IUMLThing extends IBox {
    void addAttribute(String attribute);

    void delAttribute(String attribute);

    ImmutableList<String> getAttributes();

    void addMethod(String method);

    void delMethod(String method);

    ImmutableList<String> getMethods();
}

File: sketches/model/src/main/kotlin/edu/kit/kastel/lissa/sketches/model/elements/uml/IUMLThing.kt
package edu.kit.kastel.lissa.sketches.model.elements.uml

import edu.kit.kastel.lissa.sketches.model.elements.IBox

interface IUMLThing : IBox {
    fun addAttribute(attribute: String)
    fun delAttribute(attribute: String)
    fun attributes(): List<String>
    fun addMethod(method: String)
    fun delMethod(method: String)
    fun methods(): List<String>
}